python_modules/libraries/dagster-fivetran/dagster_fivetran/translator.py

```diff
@@ -0,0 +1,97 @@
+from enum import Enum
+from typing import Any, Mapping, NamedTuple, Optional, Sequence
+
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.asset_spec import AssetSpec
+from dagster._record import record
+from dagster._serdes.serdes import whitelist_for_serdes
+
+from dagster_fivetran.utils import metadata_for_table
+
+
+class FivetranConnectorTableProps(NamedTuple):
+    table: str
+    connector_id: str
+    name: str
+    connector_url: str
+    schemas: Mapping[str, Any]
+    database: Optional[str]
+    service: Optional[str]
+
+
+class FivetranContentType(Enum):
+    """Enum representing each object in Fivetran's ontology."""
+
+    CONNECTOR = "connector"
+    DESTINATION = "destination"
+
+
+@whitelist_for_serdes
+@record
```

Comment:
Do we still want/need this class, `FivetranContentType` etc if it's not getting fed into the translator?

In particular I think having the content type isn't necessary, since we aren't exposing this data to the user directly

Reply:
I think it's worth keeping `FivetranContentType` for a few reasons:

- The content data for Fivetran is a JSON object returned by the API for both connectors and destinations. Using `FivetranContentType`, it's easier to keep track of the type of the `FivetranContentData`.
  - This pattern is used for Power BI and Tableau, where the content data is also a JSON object returned by an API.
- The implementation of `fetch_fivetran_workspace_data` leverages the `FivetranContentType` enum to create the `FivetranWorkspaceData`.
  - We use `FivetranContentType` when creating `FivetranContentData` objects and the `FivetranWorkspaceData` object, to differentiate connectors and destinations. See usage in subsequent PR, in https://github.com/dagster-io/dagster/pull/25788 and https://github.com/dagster-io/dagster/pull/25797
  
Overall, I'm mostly in favor of keeping this to keep our design pattern for `XWorkspaceData` and `fetch_x_workspace_data` as consistent as possible across integrations.